syntax = "proto2";

package isg_ai;

// instructions on creation and compiling protobuf
// https://developers.google.com/protocol-buffers/docs/pythontutorial
// compile with "protoc -I=./ --python_out=./ ./isg_ai.proto"

message ImageMaskPair {
  optional int32 channels = 1;
  optional int32 img_height = 2;
  optional int32 img_width = 3;

  optional string img_type = 4;
  optional string mask_type = 5;

  // the actual image data, in bytes
  optional bytes image = 6;

  // the actual mask data, in bytes
  optional bytes mask = 7;
  // list of labels present in the mask
  optional bytes labels = 8;
}